src/systemd-swap.py

```diff
@@ -166,13 +166,23 @@ def __init__(self, config: Config, sem: sysv_ipc.Semaphore):
         # Create parent directories for swapfc_path.
         makedirs(os.path.dirname(self.swapfc_path))
         self.fs_type, subvolume = self.get_fs_type()
-        if self.fs_type == "btrfs" and self.subvolume_possible(subvolume):
-            subprocess.run(
-                ["btrfs", "subvolume", "create", self.swapfc_path],
-                check=True,
-                stdout=subprocess.DEVNULL,
-                stderr=subprocess.DEVNULL,
-            )
+        if self.fs_type == "btrfs":
+            if not subvolume:
```

Comment:
```suggestion
        if self.fs_type == "btrfs" and not subvolume:

Reply:
The code is already correct as is. I checked and `subvolume` can only be true if the actual path (not just the parent directory) exists. We don't need to create the directory in that case, so I'm making sure we won't run that code if the file system is btrfs. I also fell for that when I was writing the patch.